doc: Connectors Data Protection (#4938)
kurrent-io/EventStore: docs/server/features/connectors/features.md

::: note
Some connectors have their own resilience mechanisms and configurations. Refer to the specific connector's page for details.
:::

> ## Data Protection

Reviewer: ```suggestion
## Data protection
```

KurrentDB Connectors implement automatic data protection for sensitive
configuration fields. This security feature ensures that confidential
information, such as passwords and access tokens, is encrypted during
transmission.

> When you configure a connector, certain sensitive fields are automatically encrypted:

Reviewer: ```suggestion
When you configure a connector, the system automatically encrypts certain sensitive fields:
```

> - The system automatically identifies sensitive keys in the configuration using predefined patterns determined internally

Reviewer: ```suggestion
- It identifies sensitive keys in the configuration using predefined internal patterns
```

> When you configure a connector, certain sensitive fields are automatically encrypted:
> 
> - The system automatically identifies sensitive keys in the configuration using predefined patterns determined internally
> - These values are encrypted using a token provided by you
> - The encrypted values are stored securely in KurrentDB instead of plaintext
> - This protection happens automatically without user configuration

Reviewer: ```suggestion
When you configure a connector, the system identifies sensitive keys using predefined internal patterns. It encrypts these keys with a token that you provide. The encrypted data is stored securely in KurrentDB rather than in plaintext. This protection is applied automatically without additional configuration.
```

> To configure data protection, you need to set the `token` property in the KurrentDB configuration file. This token is used to encrypt and decrypt sensitive data.

Reviewer: How do you set the token?

Reviewer: ```suggestion
To configure data protection, set the `token` property in the KurrentDB configuration file. The token encrypts and decrypts sensitive data.
```

> Refer to the [individual documentation](./sinks/) for each connector to see which fields are _protected_.

Reviewer: ```suggestion
Refer to each connector's [individual documentation](./sinks/) to identify which fields are _protected_.

```

```yaml
Connectors:
  Enabled: false
  DataProtection:
    Token: secret-token

> ```

Reviewer: This field is kinda lost in the page. Maybe we can rework to say that this is the configuration that you must change for the feature to be enabled. 

If that's the case, consider the following:

```
### Configuring your user token

To enable data protection, set the token property in the configuration file. The token encrypts and decrypts sensitive data.

Update your configuration file as follows:

```yaml
Connectors:
  Enabled: false
  DataProtection:
    Token: <your-token>
```
```

::: note
Some connectors have their own resilience mechanisms and configurations. Refer to the specific connector's page for details.
:::

> ## Data Protection

Reviewer 2: does this differ from the updates to this section in [PR#4943](https://github.com/kurrent-io/EventStore/pull/4943/files#diff-593c5524d8545bfb632a443ef12db35b0482d78be9f990c1a1dd779055819c52R164)?

Refer to each connector's [individual documentation](./sinks/) to see which fields are considered sensitive and protected.

### Configuring Data Protection

To enable data protection, configure the data protection settings in your
configuration file. You can provide an encryption token using either a token
file or by specifying the token directly in the configuration.

#### Using a Token File
```yaml
Connectors:
  Enabled: true
  DataProtection:
    TokenFile: "/path/to/token/file"
```

#### Using a Token Directly
```yaml
Connectors:
  Enabled: true
  DataProtection:
    Token: "<your-secret-token>"
```

::: note
If you provide both `Token` and `TokenFile`, the system will use the token file and ignore the `Token` setting.
:::

### Key Vault Configuration

KurrentDB Connectors currently only supports the Surge key vault for storing
encryption keys:

The Surge key vault is KurrentDB's native key storage mechanism that stores
encryption keys directly within KurrentDB itself. Rather than requiring a
separate external key management system, Surge stores the encrypted keys in

> internal system streams. More key vault options will be available in the future.

Reviewer: hey @w1am. I know this has already been merged, but it's generally a good idea to avoid mentioning upcoming features.
